Branch-SDK/Branch-SDK/Requests/BranchLoadRewardsRequest.m

```diff
@@ -46,18 +46,26 @@ - (void)processResponse:(BNCServerResponse *)response error:(NSError *)error {
     NSArray *responseKeys = [response.data allKeys];
     NSArray *storedKeys = [currentCreditDictionary allKeys];
 
-    if ([responseKeys count]) {
+    if ([responseKeys count] && ([response.data isKindOfClass:[NSDictionary class]] || [response.data isKindOfClass:[NSMutableDictionary class]])) {
+        
         for (NSString *key in response.data) {
-             NSInteger credits = [response.data[key] integerValue];
+            if (![key isKindOfClass:[NSString class]]) { continue; }
+            
+            NSInteger credits = [preferenceHelper getCreditCountForBucket:key];
```

Comment:
got it. i guess this has no impact on `hasUpdated` which is all i cared about. certainly an edge case.